cmake/sysbuild/partition_manager.cmake

```diff
@@ -0,0 +1,702 @@
+  --regions ${regions}
+  --output-partitions ${pm_out_partition_file}
+  --output-regions ${pm_out_region_file}
+  ${dynamic_partition_argument}
+  ${static_configuration}
+  ${region_arguments}
+  )
+
+set(pm_output_cmd
+  ${PYTHON_EXECUTABLE}
+  ${ZEPHYR_NRF_MODULE_DIR}/scripts/partition_manager_output.py
+  --input-partitions ${pm_out_partition_file}
+  --input-regions ${pm_out_region_file}
+  --config-file ${pm_out_dotconf_file}
+  )
+
+# Run the partition manager algorithm.
+execute_process(
+  COMMAND
+  ${pm_cmd}
+  RESULT_VARIABLE ret
+  )
+
+if(NOT ${ret} EQUAL "0")
+  message(FATAL_ERROR "Partition Manager failed, aborting. Command: ${pm_cmd}")
+endif()
+
+# Produce header files and config file.
+execute_process(
+  COMMAND
+  ${pm_output_cmd}
+  RESULT_VARIABLE ret
+  )
+
+if(NOT ${ret} EQUAL "0")
+  message(FATAL_ERROR "Partition Manager output generation failed, aborting. Command: ${pm_output_cmd}")
+endif()
+
+# Create a dummy target that we can add properties to for
+# extraction in generator expressions.
+add_custom_target(partition_manager)
+
+# Make Partition Manager configuration available in CMake
+import_kconfig(PM_ ${pm_out_dotconf_file} pm_var_names)
+
+foreach(name ${pm_var_names})
+  set_property(
+    TARGET partition_manager
+    PROPERTY ${name}
+    ${${name}}
+    )
+endforeach()
+
+# Turn the space-separated list into a Cmake list.
+string(REPLACE " " ";" PM_ALL_BY_SIZE ${PM_ALL_BY_SIZE})
+
+# Iterate over every partition, from smallest to largest.
+foreach(part ${PM_ALL_BY_SIZE})
+  string(TOUPPER ${part} PART)
+  get_property(${part}_PM_HEX_FILE GLOBAL PROPERTY ${part}_PM_HEX_FILE)
+  get_property(${part}_PM_ELF_FILE GLOBAL PROPERTY ${part}_PM_ELF_FILE)
+
+  # Process container partitions (if it has a SPAN list it is a container partition).
+  if(DEFINED PM_${PART}_SPAN)
+    string(REPLACE " " ";" PM_${PART}_SPAN ${PM_${PART}_SPAN})
+    list(APPEND containers ${part})
+  endif()
+
+  # Include the partition in the merge operation if it has a hex file.
+  if(DEFINED ${part}_PM_HEX_FILE)
+    get_property(${part}_PM_TARGET GLOBAL PROPERTY ${part}_PM_TARGET)
+    list(APPEND explicitly_assigned ${part})
+  else()
+    if(${part} IN_LIST images)
+      get_shared(${part}_bin_dir  IMAGE ${part} PROPERTY ZEPHYR_BINARY_DIR)
+      get_shared(${part}_hex_file IMAGE ${part} PROPERTY KERNEL_HEX_NAME)
+      get_shared(${part}_elf_file IMAGE ${part} PROPERTY KERNEL_ELF_NAME)
+      set(${part}_PM_HEX_FILE ${${part}_bin_dir}/${${part}_hex_file})
+      set(${part}_PM_ELF_FILE ${${part}_bin_dir}/${${part}_elf_file})
+      set(${part}_PM_TARGET ${part}_subimage)
+    elseif(${part} IN_LIST containers)
+      set(${part}_PM_HEX_FILE ${PROJECT_BINARY_DIR}/${part}.hex)
+      set(${part}_PM_TARGET ${part}_hex)
+    endif()
+    list(APPEND implicitly_assigned ${part})
+  endif()
+endforeach()
+
+if (${is_dynamic_partition_in_domain})
+  set(merged_suffix _${DOMAIN})
+  string(TOUPPER ${merged_suffix} MERGED_SUFFIX)
```

Comment:
Could skip set and use UNDERSCORE_DOMAIN instead

Reply:
not sure the gain, cause `merged_suffix` (lower case) is used below:
https://github.com/nrfconnect/sdk-nrf/blob/8a2122b9923d956d99c2bb98747e6a1191ddbfc2/cmake/sysbuild/partition_manager.cmake#L162

Of course that one could then also use the `_${DOMAIN}` at that location, but in this case where both lower case and upper case versions are in use later, then I think keeping code aligned with its origin makes sense.